elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metric/AbstractMetric.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.metric;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.datastores.aggregation.annotation.MetricAggregation;
+import com.yahoo.elide.datastores.aggregation.annotation.MetricComputation;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Base class of {@link Metric} that offers skeleton implementations.
+ */
+@Slf4j
+public abstract class AbstractMetric implements Metric {
+
+    /**
+     * Returns whether or not an entity field is a metric field.
+     * <p>
+     * A field is a metric field iff that field is annotated by at least one of
+     * <ol>
+     *     <li> {@link MetricAggregation}
+     *     <li> {@link MetricComputation}
+     * </ol>
+     *
+     * @param fieldName  The entity field
+     * @param containingCls  The entity that contains the {@code fieldName}
+     * @param entityDictionary  The auxiliary object that offers binding info used by this method
+     *
+     * @return {@code true} if the field is a metric field
+     */
+    public static boolean isMetricField(String fieldName, Class<?> containingCls, EntityDictionary entityDictionary) {
```

Comment:
We should try to avoid propagating EntityDictionary to too many classes.  It should remain encapsulated inside the Schema only.

Reply:
Sure.